tests/test_run_state_async.py

```
@@ -0,0 +1,55 @@
+    # though strict sorting isn't the primary thing being tested here, mostly the async execution.
+    run_ids = ["run-a", "run-b", "run-c"]
+    for run_id in run_ids:
+        run_path = runs_dir / run_id
+        run_path.mkdir()
+        state = {
+            "run_id": run_id,
+            "status": "pending",
+            "created_at": "2023-01-01T00:00:00Z",
+        }
+        (run_path / "run_state.json").write_text(json.dumps(state))
+        time.sleep(0.01)  # Ensure distinct mtimes
```

Ruff (0.14.14)

[warning] Async functions should not call `time.sleep`

(ASYNC251)